pkg/katautils/config.go

```diff
@@ -79,33 +79,36 @@ type factory struct {
 }
 
 type hypervisor struct {
-	Path                  string `toml:"path"`
-	Kernel                string `toml:"kernel"`
-	Initrd                string `toml:"initrd"`
-	Image                 string `toml:"image"`
-	Firmware              string `toml:"firmware"`
-	MachineAccelerators   string `toml:"machine_accelerators"`
-	KernelParams          string `toml:"kernel_params"`
-	MachineType           string `toml:"machine_type"`
-	BlockDeviceDriver     string `toml:"block_device_driver"`
-	EntropySource         string `toml:"entropy_source"`
-	NumVCPUs              int32  `toml:"default_vcpus"`
-	DefaultMaxVCPUs       uint32 `toml:"default_maxvcpus"`
-	MemorySize            uint32 `toml:"default_memory"`
-	MemSlots              uint32 `toml:"memory_slots"`
-	DefaultBridges        uint32 `toml:"default_bridges"`
-	Msize9p               uint32 `toml:"msize_9p"`
-	DisableBlockDeviceUse bool   `toml:"disable_block_device_use"`
-	MemPrealloc           bool   `toml:"enable_mem_prealloc"`
-	HugePages             bool   `toml:"enable_hugepages"`
-	Swap                  bool   `toml:"enable_swap"`
-	Debug                 bool   `toml:"enable_debug"`
-	DisableNestingChecks  bool   `toml:"disable_nesting_checks"`
-	EnableIOThreads       bool   `toml:"enable_iothreads"`
-	UseVSock              bool   `toml:"use_vsock"`
-	HotplugVFIOOnRootBus  bool   `toml:"hotplug_vfio_on_root_bus"`
-	DisableVhostNet       bool   `toml:"disable_vhost_net"`
-	GuestHookPath         string `toml:"guest_hook_path"`
+	Path                    string `toml:"path"`
+	Kernel                  string `toml:"kernel"`
+	Initrd                  string `toml:"initrd"`
+	Image                   string `toml:"image"`
+	Firmware                string `toml:"firmware"`
+	MachineAccelerators     string `toml:"machine_accelerators"`
+	KernelParams            string `toml:"kernel_params"`
+	MachineType             string `toml:"machine_type"`
+	BlockDeviceDriver       string `toml:"block_device_driver"`
+	EntropySource           string `toml:"entropy_source"`
+	BlockDeviceCacheSet     bool   `toml:"block_device_cache_set"`
```

Comment:
btw, I think these should have been added to some documentation and into the toml config file template - otherwise afaict no user would ever know they existed and/or how to set them - or, did I miss something?
@teawater

Reply:
Just taking a look at this, I agree we need to add this in the toml file, and document what each of these means. There is no way of knowing these even exist today. We also need tests to verify that they are picked up correctly from the toml file.

Reply:
@grahamwhaley Thanks for your remind.  I am working on it.